ibid/plugins/conversions.py

```diff
@@ -548,49 +559,63 @@ def html_flatten(soup):
         else:
             text.append(html_flatten(subsoup))
     return ''.join(text)
-    
+
 class Unihan(object):
     def __init__(self, char):
         self.char = char
         url = 'http://www.unicode.org/cgi-bin/GetUnihanData.pl?'
         params = {'codepoint': self.char.encode('utf8'),
                   'useuft8': 'true'}
         soup = get_html_parse_tree(url + urlencode(params),
-                                            treetype='html5lib-beautifulsoup')
+                                            treetype='beautifulsoup')
 
         tables = soup('table', border="1")
 
         self.data = defaultdict(unicode,
-                                ((html_flatten(td).strip() for td in row('td'))
+                                ((decode_htmlentities(html_flatten(td).strip()) for td in row('td'))
                                  for table in tables for row in table('tr')[1:]))
 
     def pinyin(self):
-        return map(fix_pinyin_tone, self.data['kMandarin'].lower().split())
+        if 'kMandarin' in self.data:
+            return map(fix_pinyin_tone, self.data['kMandarin'].lower().split())
+        return None
```

Comment:
It can make code easier to understand when we catch the exceptional cases, and return early, rather than having the main body of the function, deeply nested. (This also gives you more line width to work with.)